pkg/util/encode/encode.go

```diff
@@ -42,19 +43,24 @@ func Encode(obj runtime.Object, format string) ([]byte, error) {
 // EncodeTo converts the provided object to the specified format and
 // writes the encoded data to the provided io.Writer.
 func EncodeTo(obj runtime.Object, format string, w io.Writer) error {
-	encoder, err := EncoderFor(format)
+	encoder, err := EncoderFor(format, obj)
 	if err != nil {
 		return err
 	}
-
 	return errors.WithStack(encoder.Encode(obj, w))
 }
 
 // EncoderFor gets the appropriate encoder for the specified format.
-func EncoderFor(format string) (runtime.Encoder, error) {
+func EncoderFor(format string, obj runtime.Object) (runtime.Encoder, error) {
```

Comment:
I'm still not loving this function as-is -- if you pass it a structured object, with the `TypeMeta` filled in, and it's not one of the `vscheme` or `kscheme` types, the encoder won't produce any output. And, if it's a recognized type, without `TypeMeta`, the encoder will be for a random version for the API group (if there are multiple).

Stepping back, right now we're only ever passing this function (a) structured Velero objects, or (b) unstructured objects with the `TypeMeta` filled in.

Perhaps we should go back to the original version of this function, but with one addition - before L66 (`encoder = scheme.Codecs.EncoderForVersion(encoder, v1.SchemeGroupVersion)`), we add:

```go
if !obj.GetObjectKind().GroupVersionKind().Empty() {
    return encoder
}
```

That way, if this function is passed an object, structured or unstructured, with its `TypeMeta` filled in, it will be properly encoded, retaining its `TypeMeta`. Otherwise, it'll behave the same as it does today -- if passed a structured object that's in the velero scheme, it'll encode it; structured non-velero objects produce no output.

Reply:
By original version of the function, you mean original but with passing in a `runtime.Object`?

Reply:
yep. does this seem reasonable to you?

Reply:
Yeah, it does - I kind of like that it's requiring non-Velero objects to be explicitly set. It is, after all, a Velero encoding function.

Testing that change seemed to retain functionality on both the install command and existing commands, so I'm for using the original code and just leaving a comment documenting this behavior.